core/src/tpu.rs

```diff
@@ -153,6 +156,19 @@ impl Tpu {
             packet_sender,
             exit.clone(),
             MAX_QUIC_CONNECTIONS_PER_IP,
+            staked_nodes.clone(),
+            MAX_STAKED_CONNECTIONS,
+            MAX_UNSTAKED_CONNECTIONS,
+        )
+        .unwrap();
+
+        let tpu_forwards_quic_t = spawn_server(
```

Comment:
This will create two quic server with two runtimes. I wonder what is the trade off having the forwarder just forward to the regular Quic ports? What is the benefit of having two? I think a downside of having two probably is hard for the two quic server to make good QoS decisions. Other than this question, the changes looks good.

Reply:
The forwarded transactions are handled slightly differently than the original transactions. There's this filter that drops the forwarded transaction if the current node is not going to be the leader soon.
https://github.com/solana-labs/solana/blob/514f73f4b19ad7a584f45ec3e7a7e25233c48fd7/core/src/fetch_stage.rs#L113

This filter is important to break circular forwarding of transactions in the network.

I do agree that more work is needed to implement proper QoS handling.